### PR TITLE
Use microerror, micrologger and microendpoint

### PR DESCRIPTION
**Pull request by rossf7**
This PR

- Update operator to use microerror and micrologger
- Uses k8s client from operatorkit
- Uses  version endpoint from microendpoint

### CODE REVIEW

File: client/k8s/k8s.go [hunk 1]
```diff
@@ -1,80 +0,0 @@
-package k8s
```
Comment (rossf7) [inline client/k8s/k8s.go]: Here the old k8s client is removed

File: service/service.go [hunk 2]
```diff
@@ -71,22 +70,17 @@ func New(config Config) (*Service, error) {
...
-			BearerToken: config.Viper.GetString(config.Flag.Service.Kubernetes.BearerToken),
-			TLSClientConfig: k8sclient.TLSClientConfig{
-				CAFile:   config.Viper.GetString(config.Flag.Service.Kubernetes.TLS.CAFile),
-				CertFile: config.Viper.GetString(config.Flag.Service.Kubernetes.TLS.CrtFile),
-				KeyFile:  config.Viper.GetString(config.Flag.Service.Kubernetes.TLS.KeyFile),
-			},
-		}
-
-		k8sClient, err = k8sutil.NewClient(k8sConfig)
+		k8sConfig := k8s.DefaultConfig()
```
Comment (rossf7) [inline service/service.go]: Here the new k8sclient is used

File: service/service.go [hunk 3]
```diff
@@ -7,18 +7,17 @@ import (
 	"sync"
 
 	"github.com/giantswarm/certificatetpr"
-	microerror "github.com/giantswarm/microkit/error"
-	micrologger "github.com/giantswarm/microkit/logger"
+	"github.com/giantswarm/microendpoint/service/version"
```
Comment (rossf7) [inline service/service.go]: Here the microendpoint version service is used

**Comment by rossf7** [comment]
@kopiczko I used the script which worked great but I had to make some manual changes because the operator uses errgo.Details in various places.

e.g. https://github.com/giantswarm/aws-operator/blob/c110c6e35b70ca4e12ee61b3d178a796a94473c3/service/create/service.go#L469

File: resources/aws/errors.go [hunk 4]
```diff
@@ -28,69 +28,69 @@ const (
...
 // IsTooManyResults asserts tooManyResultsError.
 func IsTooManyResults(err error) bool {
-	return errgo.Cause(err) == tooManyResultsError
+	return microerror.Cause(err) == tooManyResultsError
 }
 
 // Delete errors.
 
-var resourceDeleteError = errgo.New("couldn't delete resource, it lacks the necessary data (ID)")
+var resourceDeleteError = microerror.New("couldn't delete resource, it lacks the necessary data (ID)")
```
Comment (xh3b4sd) [inline resources/aws/errors.go]: The error handling in the AWS operator in general is not always consistent with our other microservices. Something for boyscouting separately.

File: server/endpoint/endpoint.go [hunk 5]
```diff
@@ -1,11 +1,11 @@
 package endpoint
 
 import (
-	microerror "github.com/giantswarm/microkit/error"
-	micrologger "github.com/giantswarm/microkit/logger"
+	"github.com/giantswarm/microendpoint/endpoint/version"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
 
 	"github.com/giantswarm/aws-operator/server/endpoint/healthz"
```
Comment (xh3b4sd) [inline server/endpoint/endpoint.go]: We can also use the microendpoint health check magic.

File: service/create/service.go [hunk 6]
```diff
@@ -1066,31 +1066,31 @@ func (s *Service) deleteFunc(obj interface{}) {
 			s.logger.Log("error", errgo.Details(err))
```
Comment (xh3b4sd) [inline service/create/service.go]: We can just remove this to get rid of the errgo dependency completely by replacing `errgo.Details(err)` with `fmt.Sprintf("%#v", err)`.

File: client/k8s/k8s_test.go [hunk 7]
```diff
@@ -1,263 +0,0 @@
...
-
-import (
-	"fmt"
-	"testing"
-
-	"github.com/stretchr/testify/assert"
-	"k8s.io/client-go/rest"
-)
-
-func TestGetRawClientConfig(t *testing.T) {
```
Comment (xh3b4sd) [inline client/k8s/k8s_test.go]: Not sure if it is useful but we might want to keep some of the tests in the operatorkit lib.

File: server/endpoint/endpoint.go (hunk 5 above)
Comment (rossf7) [inline server/endpoint/endpoint.go]: Yes I'll do a follow up PR for that.

File: service/create/service.go (hunk 6 above)
Comment (rossf7) [inline service/create/service.go]: Nice :) I'll do that and we can get rid of the errgo dep.

File: client/k8s/k8s_test.go (hunk 7 above)
Comment (rossf7) [inline client/k8s/k8s_test.go]: The operatorkit client has some tests but there are more here. I'll compare and see if any need adding.

We dropped some "features" like username and password from the operatorkit client ;)

**Pull request merged**